platform/rpc/client.go

```diff
@@ -91,6 +91,20 @@ func (p *RPCClient) Version() (string, error) {
 	return version, err
 }
 
+// Export is used to get service configuration in platform-specific format
+func (p *RPCClient) Export(ids []flux.ServiceID) (string, error) {
+	var config string
+	err := p.client.Call("RPCServer.Export", ids, &config)
+	if _, ok := err.(rpc.ServerError); !ok && err != nil {
+		return "", platform.FatalError{err}
+	} else if err != nil && err.Error() == "rpc: can't find method RPCServer.Export" {
```

Comment:
Yes that was a bit vague, sorry. I meant,

 - that there's a new interface under platform that includes this extra method (and possibly other changes);
 - daemons that support the new interface connect to a new websocket endpoint, and are represented on the server side by a new implementation;
 - daemons that only support the old interface connect to the old endpoint, and are represented on the server side by an implementation that adapts where possible and returns an error otherwise.

This is a lot of trouble to go to, to avoid this one special case! But I expect more changes of this kind, and it would be good to get scaffolding in at some point. If not here, then very soon.

Also, failing the RPC is a fatal error and will kill the websocket connection. Probably not a big problem in practice, since it will reconnect, but unnecessarily messy.